Realm/Realm/Handles/SharedRealmHandle.cs

```diff
@@ -225,6 +245,72 @@ public void WriteCopy(string path, byte[] encryptionKey)
             nativeException.ThrowIfNecessary();
         }
 
+        public IntPtr CreateObject(TableHandle table)
+        {
+            NativeException ex;
+            var result = NativeMethods.create_object(this, table, out ex);
+            ex.ThrowIfNecessary();
+            return result;
+        }
+
+        public IntPtr CreateObjectWithPrimaryKey(Property pkProperty, object primaryKey, TableHandle table, string parentType, bool update, out bool isNew)
+        {
+            switch (pkProperty.Type)
+            {
+                case PropertyType.String:
+                    if (primaryKey == null)
+                    {
+                        throw new ArgumentNullException(nameof(primaryKey), "Object identifiers cannot be null");
```

Comment:
Perhaps I'm misunderstanding the logic of this function, but it looks like it's rejecting null strings as primary keys, but is completely happy with null integers for primary keys? Is that intentional?